kratos/solving_strategies/schemes/residual_based_bdf_custom_scheme.h

```diff
@@ -347,12 +333,12 @@ class ResidualBasedBDFCustomScheme
     ///@name Protected member Variables
     ///@{
 
-    std::vector<Variable< double>> mDoubleVariable;                     /// The double variables
-    std::vector<Variable< double>> mFirtsDoubleDerivatives;             /// The first derivative double variable to compute
-    std::vector<Variable< double>> mSecondDoubleDerivatives;            /// The second derivative double variable to compute
-    std::vector<Variable<array_1d<double, 3>>> mArrayVariable;          /// The array variables to compute
-    std::vector<Variable<array_1d<double, 3>>> mFirtsArrayDerivatives;  /// The first derivative array variable to compute
-    std::vector<Variable<array_1d<double, 3>>> mSecondArrayDerivatives; /// The second derivative array variable to compute
+    std::vector<Variable<double>> mDoubleVariable;                         /// The double variables
+    std::vector<Variable<double>> mFirtsDoubleDerivatives;                 /// The first derivative double variable to compute
+    std::vector<Variable<double>> mSecondDoubleDerivatives;                /// The second derivative double variable to compute
+    std::vector<VariableComponent<ComponentType>> mArrayVariable;          /// The array variables to compute
```

Comment:
mmm, i would avoid creating a copy of the variable and use a pointer instead.
as i recall (@pablobecker @pooyan-dadvand) this lead to some very subtle problems not long ago.

Reply:
True, I was the one with the problems in fact